lib/types/subdocument.js

```diff
@@ -17,6 +17,7 @@ module.exports = Subdocument;
  */
 
 function Subdocument(value, fields, parent, skipId, options) {
+  this.$__.isSubDocument = true;
```

Comment:
The tests are all blowing up because of this line. You would need to put `this.$__.isSubDocument = true` **after** `Document.call(this, value, fields, skipId, options);`, because that `Document.call()` is what creates the `.$__` property.